Framework/Algorithms/src/GetEiMonDet2.cpp

```diff
@@ -425,7 +425,11 @@ void GetEiMonDet2::parseIndices(std::vector<size_t> &detectorIndices,
         spectraListParser.parse(getProperty(PropertyNames::DETECTORS)));
     auto back = std::back_inserter(detectorIndices);
     std::copy(detectors.begin(), detectors.end(), back);
-    monitorIndex = getProperty(PropertyNames::MONITOR);
+    int monitor = getProperty(PropertyNames::MONITOR);
+    if (monitor < 0) {
```

Comment:
The code duplication was partially due to some type magick --- I think I was trying to be too smart here. Moving the `getProperty()` calls out of the `if`s does indeed simplify the code a bit.

The reason for the string *Detectors* property is that since the string is parsed by `UserStringParser`, users can specify complex ranges like `"1, 10-15, 78, 88, 100-110"`.